hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -945,7 +949,31 @@ else if ( parentModelPart instanceof EntityCollectionPart ) {
 			}
 			return false;
 		}
-		if ( cardinality == Cardinality.MANY_TO_ONE ) {
+		else if ( parentNavigablePath.getParent() != null && creationState.resolveModelPart( parentNavigablePath.getParent() ) instanceof EmbeddedCollectionPart ) {
+			/*
+				class EntityA{
+					@OneToOne(mappedBy = "identicallyNamedAssociation", fetch = FetchType.EAGER)
```

Comment:
Shouldn't this example rather use `mappedBy = "embeddable.identicallyNamedAssociation"`? Also, is there a test for this example? I kind of have the feeling that your parent check `declaringTableGroupProducer.getNavigableRole().getLocalName().equals( oneToOne.getReferencedEntityName() )` should be already enough and that we don't need this extra branch, or am I missing something?

Reply:
i don't think we support `mappedBy = "embeddable.identicallyNamedAssociation"`

Reply:
the test has a `mappedby` referring to the  [association](https://github.com/hibernate/hibernate-orm/pull/5863/files#diff-fd953c2b59c78a090f7baad15ed677fcc41d15fbd3bb169f9592b90c6b7c8d65R151) not the of the embeddable.
Without the extra checking the test fails at [assertThat( a2.getB() ).isEqualTo( b2 )](https://github.com/hibernate/hibernate-orm/pull/5863/files#diff-fd953c2b59c78a090f7baad15ed677fcc41d15fbd3bb169f9592b90c6b7c8d65R68)

Reply:
>i don't think we support mappedBy = "embeddable.identicallyNamedAssociation"

Hmm, we should though IMO 😄 

>the test has a mappedby referring to the association not the of the embeddable.

Ok, I didn't realize that this is a separate method. What happens when the association is part of an embeddable within an element collection though? What I'm trying to get to is, I think this needs a "recursive" implementation that checks parents, continuing the search when an embeddable is encountered.

Reply:
> i don't think we support mappedBy = "embeddable.identicallyNamedAssociation"

We actually do, I've used that in ORM 5.

ORM 5 even supports this with an element collection, e.g. `mappedBy = "elementCollection.identicallyNamedAssociation"`...

Reply:
Well there's this example in the javadoc of `@OneToOne`:

```

 
     Example 3: One-to-one association from an embeddable class to another entity.
 
     @Entity
     public class Employee {
        @Id int id;
        @Embedded LocationDetails location;
        ...
     }
 
     @Embeddable
     public class LocationDetails {
        int officeNumber;
        @OneToOne ParkingSpot parkingSpot;
        ...
     }
 
     @Entity
     public class ParkingSpot {
        @Id int id;
        String garage;
        @OneToOne(mappedBy="location.parkingSpot") Employee assignedTo;
         ... 
     } 
```

So I'd say JPA requires it...

Reply:
Also, in the same javadoc:

> The value of each identifier used with the dot notation is the name of the respective embedded field or property.

Reply:
Ugh, sorry, my bad, there are indeed multiple examples of this in the spec. I didn't remember this and I didn't check.

OK, good, so it *is* documented.

Sorry for the noise.

Reply:
added also the test proposed by Yoann